stl/inc/system_error

```diff
@@ -483,18 +485,14 @@ public:
     }
 
     _NODISCARD virtual string message(int _Errcode) const override {
-        const unsigned long _Size = 32767;
-        string _Narrow(_Size, '\0');
-
-        const unsigned long _Val = _Winerror_message(static_cast<unsigned long>(_Errcode), &_Narrow[0], _Size);
-        if (_Val == 0) {
-            _Narrow = "unknown error";
+        const char* _Msg     = nullptr;
+        const size_t _Length = _Winerror_message2(static_cast<unsigned long>(_Errcode), &_Msg);
+        if (_Length != 0) {
+            const unique_ptr<const char[], decltype(&_Winerror_message2_free)> _Ptr(_Msg, &_Winerror_message2_free);
+            return string(_Msg, _Length);
         } else {
-            _Narrow.resize(_Val);
+            return "unknown error";

```

Comment:
Preexisting, this pays for an unnecessary strlen.

Reply:
Will fix

Reply:
Resolved, not sure what was the preferred way so I went with a `static constexpr char[]` and `std::size`